includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -166,6 +167,34 @@ public function __construct() {
 
 		// Update the current request logged_in cookie after a guest user is created to avoid nonce inconsistencies.
 		add_action( 'set_logged_in_cookie', [ $this, 'set_cookie_on_current_request' ] );
+
+		add_filter( 'woocommerce_available_payment_gateways', [ $this, 'get_available_payment_gateways' ] );
+	}
+
+	/**
+	 * Include Multibanco in the list of payment methods.
+	 * As we are not registering Multibanco to show in the payment settings page and Multibanco is not a UPE payment method,
+	 * we need to include it here separately so that it's available in the checkout, pay for order, add payment method etc. pages.
+	 *
+	 * @param WC_Payment_Gateway[] $gateways A list of all available gateways on the payments settings page.
+	 * @return WC_Payment_Gateway[]          The same list if Multibanco is disabled or a list including the Multibanco method.
+	 */
+	public function get_available_payment_gateways( $gateways ) {
+		$stripe_index          = array_search( 'stripe', array_keys( $gateways ), true );
+		$gateways_upto_stripe  = array_slice( $gateways, 0, $stripe_index + 1 );
+		$gateways_after_stripe = array_slice( $gateways, $stripe_index + 1 );
+
+		$multibanco_gateway = new WC_Gateway_Stripe_Multibanco();
```

Comment:
Instantiating a new instance of the Multibanco class here is causing all of the hooks inside the __construct to be registered multiple times, causing things like this on the thank-you page and probably other issues: 
![image](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/2275145/19284c41-d068-46d2-99cb-9274beda84f0)